http/rest-client-reactive/src/test/java/io/quarkus/ts/http/restclient/reactive/MultipartPostEncodeIT.java

```diff
@@ -0,0 +1,103 @@
+package io.quarkus.ts.http.restclient.reactive;
+
+import static org.hamcrest.Matchers.is;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.nio.file.Paths;
+
+import org.jboss.logging.Logger;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import io.quarkus.test.bootstrap.RestService;
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.quarkus.test.services.QuarkusApplication;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.http.HttpServer;
+
+@QuarkusScenario
+public class MultipartPostEncodeIT {
+
+    private static final Logger LOGGER = Logger.getLogger(MultipartPostEncodeIT.class);
+    private static final String FILE = "file";
+    private static final String TEXT = "text";
+    private static Vertx vertx;
+    private static Closeable serverCloseable;
+    private static String detectedEncoderMode;
+
+    @BeforeAll
+    public static void setUp() {
+        vertx = Vertx.vertx();
+        serverCloseable = createHttpServer(vertx);
+    }
+
+    private static Closeable createHttpServer(Vertx vertx) {
+        HttpServer httpServer = vertx.createHttpServer();
+        httpServer.requestHandler(req -> {
+            if (req.method() == HttpMethod.POST
+                    && (req.path().equals("/multipart/upload") || req.path().equals("/encode/upload"))) {
+                req.setExpectMultipart(true);
+
+                String contentType = req.getHeader("Content-Type");
+                if (contentType.startsWith("multipart/form-data")) {
+                    detectedEncoderMode = "HTML5";
+                } else if (contentType.startsWith("application/x-www-form-urlencoded")) {
+                    String body = req.formAttributes().get(TEXT);
+                    if (body.contains("+") || body.contains("=")) {
+                        detectedEncoderMode = "RFC1738";
+                    } else if (body.contains("%2B") || body.contains("%3D")) {
+                        detectedEncoderMode = "RFC3986";
+                    } else {
+                        detectedEncoderMode = "UNKNOWN";
+                    }
+                } else {
+                    detectedEncoderMode = "UNKNOWN";
+                }
+
+                req.response()
+                        .putHeader("Content-Type", "application/json")
+                        .end("{\"status\": \"received\"}");
+            } else {
+                req.response().setStatusCode(404).end();
+            }
+        }).listen(8443)
+                .onSuccess(server -> LOGGER.info("Vert.x HTTP server started on port 8443"))
+                .onFailure(err -> {
+                    LOGGER.error("Failed to start Vert.x HTTP server", err);
+
+                });
+
+        return httpServer::close;
```

Comment:
I think you are misunderstanding `Closeable`. TBH I didn't see this before, but I think here the method reference `httpServer::close` becomes same as

```
var closeable = new Closeable {

   void close() {
     httpServer.close();
 }

};
return closeable;
```
which means you are not waiting for the Future.